room-gacha/index.ts

```diff
@@ -0,0 +1,218 @@
+}
+
+interface Room {
+    name: string;
+    url: string;
+    images: Image[];
+    layoutImageUrl: string;
+    rent: string;
+    layout: string;
+    size: string;
+    direction: string;
+    years: string;
+    address: string;
+    access: string[];
+}
+
+const pickOneResult = async (cityIDs: string[], ar: string, bs: string, ta: string) => {
+    const appAddress = 'https://suumo.jp/jj/chintai/ichiran/FR301FC005/'; // 部屋ごとに表示
+    const queries = {
+        sc: cityIDs,
+        ar, bs, ta, // 謎 (必須) (都道府県固有)
+        po01: '09', // 並び替え: 新着順
+        pc: '100',  // 表示件数: 100件
+    };
+    const url = `${appAddress}?${qs.stringify(queries)}`;
+    interface SearchResult {
+        title: string;
+        hit: string;
+        rooms: Room[];
+    }
+    const result = await scrapeIt<SearchResult>(url, {
+        title: {
+            selector: '.ui-section-header h1',
+            convert: s => s.split(' ')[0],
+        },
+        hit: '.paginate_set-hit',
+        rooms: {
+            listItem: '.property',
+            data: {
+                name: '.property_inner-title a',
+                url: {
+                    selector: '.property_inner-title a',
+                    attr: 'href',
+                    convert: s => `https://suumo.jp${s}`,
+                },
+                images: {
+                    listItem: '.cassette_carrousel-item li',
+                    data: {
+                        url: {
+                            selector: 'img',
+                            attr: 'rel',
+                        },
+                        description: {
+                            selector: 'img',
+                            attr: 'alt',
+                        },
+                    },
+                },
+                layoutImageUrl: {
+                    selector: 'img[alt=間取り]',
+                    attr: 'rel',
+                },
+                rent: '.detailbox-property-point',
+                layout: {
+                    selector: '.detailbox-property--col3 div',
+                    eq: 0,
+                },
+                size: {
+                    selector: '.detailbox-property--col3 div',
+                    eq: 1,
+                },
+                direction: {
+                    selector: '.detailbox-property--col3 div',
+                    eq: 2,
+                },
+                years: {
+                    selector: '.detailbox-property--col3 div',
+                    eq: 4,
+                },
+                address: {
+                    selector: '.detailbox-property-col',
+                    eq: 4,
+                },
+                access: {
+                    selector: '.detailnote-box',
+                    eq: 0,
+                    convert: s => s.split('\n').map((e: string) => e.trim()),
+                },
+            },
+        },
+    });
+    const searchResult = result.data;
+    const pickedRoom = _.shuffle(searchResult.rooms)[0];
```

Comment:
[may] `_.sample`でスッキリ書けます

```suggestion
    const pickedRoom = _.sample(searchResult.rooms);
```

Reply:
天才